attributecert/attributecert.go

```diff
@@ -0,0 +1,712 @@
+	PropertyName  string
+	PropertyValue string
+	Status        asn1.Enumerated `asn1:"optional,tag=0"`
+}
+
+type AttributeCertificateIdentifier struct {
+	HashAlgorithm          pkix.AlgorithmIdentifier
+	HashOverSignatureValue string
+}
+
+type CertificateIdentifier struct {
+	AttributeCertIdentifier AttributeCertificateIdentifier `asn1:"optional,tag=0"`
+	GenericCertIdientifier  issuerSerial                   `asn1:"optional,tag=1"`
+}
+
+type ComponentAddress struct {
+	AddressType  asn1.ObjectIdentifier
+	AddressValue string
+}
+
+type ComponentClass struct {
+	ComponentClassRegistry asn1.ObjectIdentifier
+	ComponentClassValue    string
+}
+
+type ComponentIdentifierV2 struct {
+	ComponentClass           ComponentClass
+	ComponentManufacturer    string
+	ComponentModel           string
+	ComponentSerial          string                `asn1:"optional,tag=0"`
+	ComponentRevision        string                `asn1:"optional,tag=1"`
+	ComponentManufacturerId  int                   `asn1:"optional,tag=2"`
+	FieldReplaceable         bool                  `asn1:"optional,tag=3"`
+	ComponentAddresses       []ComponentAddress    `asn1:"optional,tag=4"`
+	ComponentPlatformCert    CertificateIdentifier `asn1:"optional,tag=5"`
+	ComponentPlatformCertUri string                `asn1:"optional,tag=6"`
+	Status                   asn1.Enumerated       `asn1:"optional,tag=7"`
+}
+
+type UriReference struct {
+	UniformResourceIdentifier string
+	HashAlgorithm             pkix.AlgorithmIdentifier `asn1:"optional"`
+	HashValue                 string                   `asn1:"optional"`
+}
+
+type PlatformConfigurationV2 struct {
+	ComponentIdentifiers    []ComponentIdentifierV2 `asn1:"optional,tag=0"`
+	ComponentIdentifiersUri UriReference            `asn1:"optional,tag=1"`
+	PlatformProperties      []Property              `asn1:"optional,tag=2"`
+	PlatformPropertiesUri   UriReference            `asn1:"optional,tag=3"`
+}
+
+type ComponentIdentifierV1 struct {
+	ComponentManufacturer   string
+	ComponentModel          string
+	ComponentSerial         string             `asn1:"optional,tag=0"`
+	ComponentRevision       string             `asn1:"optional,tag=1"`
+	ComponentManufacturerId int                `asn1:"optional,tag=2"`
+	FieldReplaceable        bool               `asn1:"optional,tag=3"`
+	ComponentAddresses      []ComponentAddress `asn1:"optional,tag=4"`
+}
+
+type PlatformConfigurationV1 struct {
+	ComponentIdentifiers  []ComponentIdentifierV1 `asn1:"optional,tag=0"`
+	PlatformProperties    []Property              `asn1:"optional,tag=1"`
+	PlatformPropertiesUri []UriReference          `asn1:"optional,tag=2"`
+}
+
+func unmarshalSAN(v asn1.RawValue) ([]pkix.AttributeTypeAndValue, error) {
+	var attributes []pkix.AttributeTypeAndValue
+	if v.Tag == asn1.TagSet {
+		var e pkix.AttributeTypeAndValue
+		_, err := asn1.Unmarshal(v.Bytes, &e)
+		if err != nil {
+			return nil, err
+		}
+		attributes = append(attributes, e)
+		return attributes, nil
```

Comment:
Instead of appending to an always-empty slice, could we just return this as a literal value?

```go
return []pkix.AttributeTypeAndValue{e}, nil
```